eng/common/scripts/Verify-RestApiSpecLocation.ps1

```diff
@@ -227,12 +222,24 @@ function Verify-PackageVersion() {
   }
 }
 
-try{
+try {
   # Verify package version is not a prerelease version, only continue the validation if the package is GA version
   Verify-PackageVersion
 
   $ServiceDir = Join-Path $RepoRoot 'sdk' $ServiceDirectory
```

Comment:
We don't need to build this path. You already have the information from the pkgInfo object you create in `Verify-PackageVersion`. That function is computing the properties and has all the logic. For example the package info object for `@azure-rest/core-client` has the following:
```
{
  "Name": "@azure-rest/core-client",
  "Version": "1.2.0",
  "DevVersion": "1.2.0-alpha.20240123.1",
  "DirectoryPath": "sdk/core/core-client-rest",
  "ServiceDirectory": "core",
  "ReadMePath": "sdk/core/core-client-rest/README.md",
  "ChangeLogPath": "sdk/core/core-client-rest/CHANGELOG.md",
  "Group": null,
  "SdkType": "client",
  "IsNewSdk": true,
  "ArtifactName": "azure-rest-core-client",
  "ReleaseStatus": "Unreleased"
}
```

So instead return that pkgInfo object and use those properties so we don't have to maintain this extra logic that will definitely get broken at some point. That is the reason we have these common package properties and have each language implement them directly.

Reply:
To add a little more context we have a common class at https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/Package-Properties.ps1#L29 that contains these properties and we have each language create this object during discover see JS's implementation for example https://github.com/Azure/azure-sdk-for-js/blob/main/eng/scripts/Language-Settings.ps1#L28 so it handles all these special edge cases.